### PR TITLE
[stable-2.8] Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI (#59557)

### LINKED ISSUE #59548: Modules in parallel library/ directory not found for imported playbooks in Ansible 2.8
##### SUMMARY

In Ansible 2.7, when I imported a playbook in another directory, custom modules in a library directory located parallel to that playbook were found, and the play executed without issue.

In Ansible 2.8, those custom modules are no longer found, and the library directory must be parallel to the root playbook for successful execution.
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

framework
##### ANSIBLE VERSION

```paste below
ansible 2.8.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cloud-user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Tested on RHEL 7.5

##### STEPS TO REPRODUCE

https://gist.github.com/sasjowood/545c7969cdc30d41a9b9ff6a8c3106c9

Using files from the gist link above, create the following structure:

├── my_plays
│   ├── library
│   │   └── my_custom_module.py
│   └── my_play.yml
└── site.yml

Run `ansible-playbook site.yml`

```yaml

```

##### EXPECTED RESULTS

Results with Ansible 2.7.10:

```paste below
ansible-playbook 2.7.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cloud-user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
Parsed /etc/ansible/hosts inventory source with ini plugin
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: site.yml ******************************************************************************************************
2 plays in site.yml

PLAY [Main play] ******************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************
task path: /home/cloud-user/ansible-test/site.yml:1
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: cloud-user
<127.0.0.1> EXEC /bin/sh -c 'echo ~cloud-user && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178 `" && echo ansible-tmp-1563994361.68-66921848860178="` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/setup.py
<127.0.0.1> PUT /home/cloud-user/.ansible/tmp/ansible-local-10284Z9N2pv/tmpGgLqPn TO /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178/AnsiballZ_setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178/ /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178/AnsiballZ_setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python2 /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178/AnsiballZ_setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/cloud-user/.ansible/tmp/ansible-tmp-1563994361.68-66921848860178/ > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers
META: ran handlers
META: ran handlers

PLAY [My Playbook] ******************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************
task path: /home/cloud-user/ansible-test/my_plays/my_play.yml:1
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: cloud-user
<127.0.0.1> EXEC /bin/sh -c 'echo ~cloud-user && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733 `" && echo ansible-tmp-1563994362.64-268460030813733="` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/system/setup.py
<127.0.0.1> PUT /home/cloud-user/.ansible/tmp/ansible-local-10284Z9N2pv/tmpR7lKMt TO /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733/AnsiballZ_setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733/ /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733/AnsiballZ_setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python2 /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733/AnsiballZ_setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/cloud-user/.ansible/tmp/ansible-tmp-1563994362.64-268460030813733/ > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [my_custom_module] ******************************************************************************************************
task path: /home/cloud-user/ansible-test/my_plays/my_play.yml:5
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: cloud-user
<127.0.0.1> EXEC /bin/sh -c 'echo ~cloud-user && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376 `" && echo ansible-tmp-1563994363.13-100245842986376="` echo /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376 `" ) && sleep 0'
Using module file /home/cloud-user/ansible-test/my_plays/library/my_custom_module.py
<127.0.0.1> PUT /home/cloud-user/.ansible/tmp/ansible-local-10284Z9N2pv/tmpXgopiI TO /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376/AnsiballZ_my_custom_module.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376/ /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376/AnsiballZ_my_custom_module.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python2 /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376/AnsiballZ_my_custom_module.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/cloud-user/.ansible/tmp/ansible-tmp-1563994363.13-100245842986376/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "msg": "Hello, world!"
        }
    },
    "my_custom_module_msg": "Hello, world!"
}
META: ran handlers
META: ran handlers

PLAY RECAP ******************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0

```

##### ACTUAL RESULTS

Results with Ansible 2.8.*:

```paste below
ansible-playbook 2.8.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/cloud-user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
Parsed /etc/ansible/hosts inventory source with ini plugin
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to be in '/home/cloud-user/ansible-test/my_plays/my_play.yml': line 5, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - my_custom_module:
      ^ here
```

### PR DESCRIPTION
**Pull request by sivel**
* Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI. Fixes #59548

* Restore cli additions
(cherry picked from commit 923e218)

Co-authored-by: Matt Martz <matt@sivel.net>

### CODE REVIEW

File: changelogs/fragments/59548-import-playbook-plugin-dirs.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+- plugin loader - Move plugin loader playbook dir additions back to ``Playbook`` instead of ``PlaybookCLI``
```
Comment (abadger) [inline changelogs/fragments/59548-import-playbook-plugin-dirs.yml]: From the terminology here (Move), it sounds like the code was removed from PlaybookCLI and added to Playbook.  I see that that was what was originally proposed but that it seems to have intentionally changed to be "in addition to" instead of moved.  If it's okay with you, I'll change this changelog entry to reflect that.

Also, I couldn't find the reason that the solution was changed to be used in both cases.  Just for my own curiousity, could you summarize that?

Comment (abadger) [reply changelogs/fragments/59548-import-playbook-plugin-dirs.yml]: https://github.com/ansible/ansible/pull/60330

Comment (sivel) [reply changelogs/fragments/59548-import-playbook-plugin-dirs.yml]: Yes, originally I was moving it, and didn't update the fragment later when I determined that it needed to exist in 2 places.

Changes were made in the become plugins PR to how/when plugins were loaded.  When I tested with a playbook relative inventory plugin (which is newly supported), loading it in the initial playbook via `Playbook.__init__` was too late, so it needed to be done in both the CLI and in the playbook.  This has the side effect of not being able to support inventory plugins from included playbook dirs if they are different than the parent, but that isn't a regression.

**Comment by abadger** [comment]
Merged for the 2.8.4 release.
